src/components/victory-axis/helper-methods.js

```diff
@@ -178,8 +179,9 @@ export default {
     const verticalAnchor = sign < 0 ? "end" : "start";
     const labelStyle = style.axisLabel;
     return {
-      x: x + globalTransform.x,
-      y: (sign * labelPadding) + globalTransform.y,
+      x: x - (globalTransform.x / vPadding),
```

Comment:
maybe try taking the values from the axis line positions, since we know those are ending up in the right place